File: CS/Sample/Controllers/HomeController.cs
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using Sample.Models;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Sample.Controllers {
    public class HomeController : Controller {
        PersonsList list = new PersonsList();

        public ActionResult Index() {
            FillViewBag();
            return View(list.GetPersons());
        }

        public ActionResult GridViewEditingPartial() {
            FillViewBag();
            return PartialView(list.GetPersons());        
        }

        private void FillViewBag()
        {
            ViewBag.Roles = new List<Role>()
            {
                new Role(){ RoleID = 1, RoleName = "Role 1"},
                new Role(){ RoleID = 2, RoleName = "Role 2"},
                new Role(){ RoleID = 3, RoleName = "Role 3"},
                new Role(){ RoleID = 4, RoleName = "Role 4"},
                new Role(){ RoleID = 5, RoleName = "Role 5"}
            };
        }


        [HttpPost, ValidateInput(false)]
        public ActionResult EditingAddNew([ModelBinder(typeof(DevExpressEditorsBinder))] Person person) {
            if (ModelState.IsValid)
                list.AddPerson(person);
            FillViewBag();
            return PartialView("GridViewEditingPartial", list.GetPersons());
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult EditingUpdate(Person personInfo) {
            string[] tokens = TokenBoxExtension.GetSelectedValues<string>("Roles");
            personInfo.Roles = string.Join(",", tokens);
            if (ModelState.IsValid)
                list.UpdatePerson(personInfo);
            FillViewBag();
            return PartialView("GridViewEditingPartial", list.GetPersons());
        }

        public ActionResult EditingDelete(int personId) {
            list.DeletePerson(personId);
            FillViewBag();
            return PartialView("GridViewEditingPartial", list.GetPersons());
        }
    }
}

File: VB/Sample/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports Sample.Models
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace Sample.Controllers
    Public Class HomeController
        Inherits Controller

        Private list As New PersonsList()

        Public Function Index() As ActionResult
            FillViewBag()
            Return View(list.GetPersons())
        End Function

        Public Function GridViewEditingPartial() As ActionResult
            FillViewBag()
            Return PartialView(list.GetPersons())
        End Function

        Private Sub FillViewBag()
            ViewBag.Roles = New List(Of Role)() From { _
                New Role() With {.RoleID = 1, .RoleName = "Role 1"}, _
                New Role() With {.RoleID = 2, .RoleName = "Role 2"}, _
                New Role() With {.RoleID = 3, .RoleName = "Role 3"}, _
                New Role() With {.RoleID = 4, .RoleName = "Role 4"}, _
                New Role() With {.RoleID = 5, .RoleName = "Role 5"} _
            }
        End Sub


        <HttpPost, ValidateInput(False)> _
        Public Function EditingAddNew(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal person As Person) As ActionResult
            If ModelState.IsValid Then
                list.AddPerson(person)
            End If
            FillViewBag()
            Return PartialView("GridViewEditingPartial", list.GetPersons())
        End Function

        <HttpPost, ValidateInput(False)> _
        Public Function EditingUpdate(ByVal personInfo As Person) As ActionResult
            Dim tokens() As String = TokenBoxExtension.GetSelectedValues(Of String)("Roles")
            personInfo.Roles = String.Join(",", tokens)
            If ModelState.IsValid Then
                list.UpdatePerson(personInfo)
            End If
            FillViewBag()
            Return PartialView("GridViewEditingPartial", list.GetPersons())
        End Function

        Public Function EditingDelete(ByVal personId As Integer) As ActionResult
            list.DeletePerson(personId)
            FillViewBag()
            Return PartialView("GridViewEditingPartial", list.GetPersons())
        End Function
    End Class
End Namespace